web/viewer.js

```diff
@@ -1300,6 +1300,35 @@ var PDFViewerApplication = {
 
     var body = document.querySelector('body');
     body.setAttribute('data-mozPrintCallback', true);
+
+    if (!this.hasEqualPageSizes) {
+      console.warn('Not all pages have the same size. The printed result ' +
+          'may be incorrect!');
+    }
+
+    // Insert a @page + size rule to make sure that the page size is correctly
+    // set. Note that we assume that all pages have the same size, because
```

Comment:
Variable page sizes are not supported by PDF.js because browsers do not offer the means to implement this (named pages). This is explained on the next few lines, and the relevant browser bugs have been referenced.